docs/reusing-typecheck-results.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Reusing typecheck results
+category: Compiler Internals
+categoryindex: 200
+index: 42
+---
+
+# Reusing typecheck results between compiler and tooling runs
+
+Caching and reusing typecheck results between compiler and tooling runs will be an optimization technique aimed at improving the performance and efficiency of the F# development experience.
+
+### Why is it important
+
+- **Performance**. Recomputing type information for large projects is time-consuming. By caching the results, subsequent runs can skip this step, leading to faster builds.
+
+- **IDE responsiveness**. IDEs can provide faster IntelliSense, code navigation, and other features if they can quickly access cached type information.
+
+- **Error reduction**. Consistent typechecking results help avoid scenarios where slight differences between runs could cause different errors or warnings to be produced.
+
+### General considerations
+
+- The technique will be implemented separately for the compiler and tooling, for better granularity and trackability.
+
+- Caching data will likely be unsophisticated, the harder part will be to understand what data can be reused and how.
```

Comment:
I would say choosing format for serialisation might be challenging, since we need it to be iterative